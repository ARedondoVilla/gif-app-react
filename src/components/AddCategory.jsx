import React, { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const onChangeInput = (event) => {
    // console.log('>> event', event.target.value);
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault(); // EVITA QUE SE ACTUALICE LA PAGINA CUANDO HAGAMOS submit
    // console.log(">> event submit", event);
    // console.log("inputValue", inputValue);
    setCategories((categories) => [inputValue, ...categories]);
    if (inputValue.trim().length <= 1) return;
    // SE MANDA EL setCategories MEDIAENTE LAS PROPS, PERO COMO NO TENEMOS LAS categories, USAMOS
    // EL CALLBACK PARA MODIFICAR EL ARRAY DE categories, USANDO EL CALLBACK ACCEDEMOS A LA INFORMACION
    // DE categories.
  };

  
  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Busca un gif"
        value={inputValue}
        // onChange={(event) => onChangeInput(event)}
        // CODIGO EQUIVALENTE, PERO EN LA DEFINCION DE LA FUNCION SE SIGUE PASANDO EL PARAMETRO event
        onChange={onChangeInput}
      />
    </form>
  );
};
