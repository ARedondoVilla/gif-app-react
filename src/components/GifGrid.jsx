import React, { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const [gifs, setGifs] = useState([]);

  const setGifList = async () => {
    const gifList = await getGifs(category)
    setGifs(gifList)
  }

  useEffect(() => {
    setGifList();
  }, []); 
  // ARRRAY DE DEPENDENCIAS => SI ESTA VACIO SE EJECUTA SOLO LA PRIMERA VEZ EL CONTENIDO DENTRO
  // DEL CALLBACK DEL useEffect, SI TIENE EL VALOR DE ALGUNA VARIABLE, SE EJECUTA LA PRIMERA VEZ Y CADA
  // VEZ QUE SE DETECTE ALGUN CAMBIO DE ESTA VARIABLE
  /////// IMPORTANTE ///////////
  // EN CONSOLA SE EJECUTA DOS VECES AL REFRESCAR LA PAGINA DEBIDO AL <React.StrictMode> QUE HAY
  // EN EL COMPONENTE main.jsx. SOLO SE LANZA DOS VECES EN DESARROLLO, EN PRODUCCION SOLO SE EJECUTA
  // UNA SOLA VEZ. SI COMENTAMOS LA ETIQUETA <React.StrictMode> DEL main.jsx SE EJECUTA SOLO UNA VEZ.

  return (
    <>
      <h3>{category}</h3>
      <ol>
        {
          gifs.map(({id, title}) => (
            <li key={id}>{title}</li>
          ))
        }
      </ol>
    </>
  );
};
