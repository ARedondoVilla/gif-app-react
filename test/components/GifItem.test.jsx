import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem Component", () => {
  const title = "Titulo";
  const url = "url";
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
