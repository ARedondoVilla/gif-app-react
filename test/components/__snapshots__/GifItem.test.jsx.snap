// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Component Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="url"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
